Add build scripts for IPOPT + dependencies

tkoskela opened the pull request:
Adds a build scripts requested in #440 for 

- IPOPT 3.13.4
- A third-party provided MUMPS 5.4.0 (with shared library objects)
- metis 5.1.0 (with shared library objects)

All are built with `intel/2018/update3` and `gcc-libs/4.9.2`

heatherkellyucl commented on ipopt-3.13.4_install:
```diff
@@ -0,0 +1,43 @@
+package_name="ipopt"
+package_version="3.13.4"
+package_description="Ipopt (Interior Point OPTimizer, pronounced eye-pea-Opt) is a software package for large-scale nonlinear optimization."
+SRC_ARCHIVE=${SRC_ARCHIVE:-git@github.com:coin-or/Ipopt.git}
```
If you are git cloning the repo rather than downloading a release tarfile and building that, you need to make sure you are getting a specific version, so if this is rerun in a year, it gets the same code.

heatherkellyucl commented on mumps-thirdparty-5.4.0_install:
```diff
@@ -0,0 +1,37 @@
+
+cd "$build_dir"
+
+
```
Same with this git clone, has to make sure it gets a specific version.

heatherkellyucl reviewed (changes requested):
The rest looks fine - I'm assuming they run correctly!

tkoskela commented:
I changed the git clones to wget'ing release tarballs from github. Also updated Ipopt to the latest version of the 3.14 release that was released recently and compiles successfully.

The Thirdparty-Mumps library has it's own internal version numbering that is separate from Mumps version numbering. I changed the module to use the Thirdparty-Mumps internal version number 3.0.0 but added a note in the module description that it's retrieving Mumps 5.4.0
